v3/pkg/application/application.go

```diff
@@ -774,9 +785,15 @@ func SaveFileDialogWithOptions(s *SaveFileDialogOptions) *SaveFileDialogStruct {
 	return result
 }
 
-func (a *App) dispatchEventToWindows(event *WailsEvent) {
+func (a *App) dispatchEventToListeners(event *WailsEvent) {
+	listeners := a.wailsEventListeners
+
 	for _, window := range a.windows {
-		window.DispatchWailsEvent(event)
+		listeners = append(listeners, window)
```

Comment:
```suggestion
		window.DispatchWailsEvent(event)
```

Reply:
Either we add the windows to the listeners when they are created or just dispatch to them explicitly. Given that windows can come and go, it's probably better to do that the original way (explicitly).